_examples/real-world-examples/server-sent-events/server/http.go

```diff
@@ -0,0 +1,318 @@
+	}
+
+	postStream := postStreamAdapter{storage: router.PostsStorage, logger: router.Logger}
+	feedStream := feedStreamAdapter{storage: router.FeedsStorage, logger: router.Logger}
+	allFeedsStream := allFeedsStreamAdapter{storage: router.FeedsStorage, logger: router.Logger}
+
+	postHandler := sseRouter.AddHandler(PostUpdatedTopic, postStream)
+	feedHandler := sseRouter.AddHandler(FeedUpdatedTopic, feedStream)
+	allFeedsHandler := sseRouter.AddHandler(FeedUpdatedTopic, allFeedsStream)
+
+	r.Route("/api", func(r chi.Router) {
+		r.Get("/posts/{id}", postHandler)
+		r.Post("/posts", router.CreatePost)
+		r.Post("/generate/post", router.GeneratePost)
+		r.Patch("/posts/{id}", router.UpdatePost)
+		r.Get("/feeds/{name}", feedHandler)
+		r.Get("/feeds", allFeedsHandler)
+	})
+
+	go func() {
+		err = sseRouter.Run(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	<-sseRouter.Running()
+
+	return r
+}
+
+type feedSummary struct {
+	Name  string `json:"name"`
+	Posts int    `json:"posts"`
+}
+
+type AllFeedsResponse struct {
+	Feeds []feedSummary `json:"feeds"`
+}
+
+type allFeedsStreamAdapter struct {
+	storage FeedsStorage
+	logger  watermill.LoggerAdapter
+}
+
+func (f allFeedsStreamAdapter) GetResponse(w http.ResponseWriter, r *http.Request) (interface{}, bool) {
+	feeds, err := f.storage.All(r.Context())
+	if err != nil {
+		logAndWriteError(f.logger, w, err)
+		return nil, false
+	}
+
+	response := AllFeedsResponse{}
+
+	for _, f := range feeds {
+		response.Feeds = append(response.Feeds, feedSummary{
+			Name:  f.Name,
+			Posts: len(f.Posts),
+		})
+	}
+
+	return response, true
+}
+
+func (f allFeedsStreamAdapter) Validate(r *http.Request, msg *message.Message) (ok bool) {
+	return true
+}
+
+type CreatePostRequest struct {
+	Title   string `json:"title"`
+	Content string `json:"content"`
+	Author  string `json:"author"`
+}
+
+func (router Router) CreatePost(w http.ResponseWriter, r *http.Request) {
+	req := CreatePostRequest{}
+	err := render.Decode(r, &req)
+	if err != nil {
+		logAndWriteError(router.Logger, w, err)
+		return
+	}
+
+	post := NewPost(
+		watermill.NewUUID(),
+		req.Title,
+		req.Content,
+		req.Author,
+	)
+
+	err = router.addPost(r.Context(), post)
+	if err != nil {
+		logAndWriteError(router.Logger, w, err)
+		return
+	}
+
+	w.WriteHeader(204)
+}
+
+func (router Router) GeneratePost(w http.ResponseWriter, r *http.Request) {
+	title := gofakeit.Sentence(5)
+	content := gofakeit.Sentence(20)
+	author := gofakeit.Name()
+
+	tagsCount := rand.Intn(3) + 1
+	for i := 0; i < tagsCount; i++ {
+		content += fmt.Sprintf(" #%v", generatedTags[rand.Intn(len(generatedTags))])
+	}
+
+	post := NewPost(
+		watermill.NewUUID(),
+		title,
+		content,
+		author,
+	)
+
+	err := router.addPost(r.Context(), post)
+	if err != nil {
+		logAndWriteError(router.Logger, w, err)
+		return
+	}
+
+	w.WriteHeader(204)
+}
+
+func (router Router) addPost(ctx context.Context, post Post) error {
+	err := router.PostsStorage.Add(ctx, post)
```

Comment:
maybe posts storage can be write model (BTW it may be good to explain difference between read model and write model in readme)

Reply:
a378fb8e009055137f551096f86dd85b0144524a